Plugins/Sycl/src/Seeding/CreateSeedsForGroupSycl.cpp

```diff
@@ -440,54 +587,106 @@ void createSeedsForGroupSycl(
+                  // space point by binary search
                   auto L = firstMiddle;
                   auto R = lastMiddle;
                   auto mid = L;
                   while (L < R - 1) {
                     mid = (L + R) / 2;
-                    if (idx + tripletSearchOffset[0] < deviceSumComb[mid]) {
+                    // To be able to search in deviceSumComb, we need
+                    // to use an offset (sumCombUptoFirstMiddle).
+                    if (idx + sumCombUptoFirstMiddle < deviceSumComb[mid]) {
                       R = mid;
                     } else {
                       L = mid;
                     }
                   }
                   mid = L;
 
-                  detail::TripletData T = {MIN, MIN};
-                  deviceCurvImpact[idx] = T;
-                  const auto numT = deviceNumTopDuplets[mid];
+                  const auto numT = numTopDupletsAcc[mid];
+                  const auto threadIdxForMiddleSP =
+                      (idx - deviceSumComb[mid] + sumCombUptoFirstMiddle);
+
+                  // NOTES ON THREAD MAPPING TO SPACE POINTS
+                  /*
+                    We need to map bottom and top SP indices to this
+                    thread. This is done in the following way: We
+                    calculated the number of possible triplet
+                    combinations for this middle SP (let it be
+                    num_comp_bot*num_comp_top). Let num_comp_bot = 2
+                    and num_comp_top=3 in this example. So we have 2
+                    compatible bottom and 3 compatible top SP for this
+                    middle SP.
+
+                    That gives us 6 threads altogether:
+                              ===========================================
+                    thread:    |  0   |  1   |  2   |  3   |  4   |  5
+                    | bottom id: | bot0 | bot0 | bot0 | bot1 | bot1 |
+                    bot1 | top id:    | top0 | top1 | top2 | top0 |
+                    top1 | top2 |
+                              ===========================================
+
+                    If we divide 6 by the number of compatible top SP
+                    for this middle SP, or deviceNumTopDuplets[mid]
+                    which is 3 now, we get the id for the bottom SP.
+                    Similarly, if we take modulo
+                    deviceNumTopDuplets[mid], we get the id for the
+                    top SP.
+
+                    So if threadIdxForMiddleSP = 3, then ib = 1 and it
+                    = 0.
+
+                    We can use these ids together with
+                    deviceSumBot[mid] and deviceSumTop[mid] to be able
+                    to index our other arrays that actually indices
+                    for corresponding bottom and top SPs (deviceIndBot
+                    and deviceIndTop) or data of duplets in linear
+                    equation form (deviceLinBot and deviceLinTop).
```

Comment:
long run-on sentence... I'm not sure what you're saying here

Reply:
I don't understand it either. It should be clearer now. :)